Box.V2/Managers/BoxMetadataCascadePolicyManager.cs

```diff
@@ -0,0 +1,142 @@
+﻿using Box.V2.Auth;
+using Box.V2.Config;
+using Box.V2.Converter;
+using Box.V2.Extensions;
+using Box.V2.Models;
+using Box.V2.Models.Request;
+using Box.V2.Services;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Box.V2.Managers
+{
+    /// <summary>
+    /// The class managing the Box API's Metadata Cascade Policies endpoint.
+    /// </summary>
+    public class BoxMetadataCascadePolicyManager : BoxResourceManager
+    {
+        /// <summary>
+        /// Create a new BoxMetadataCascadePolicy object.
+        /// </summary>
+        /// <param name="config"></param>
+        /// <param name="service"></param>
+        /// <param name="converter"></param>
+        /// <param name="auth"></param>
+        /// <param name="asUser"></param>
+        /// <param name="suppressNotifications"></param>
+        public BoxMetadataCascadePolicyManager(IBoxConfig config, IBoxService service, IBoxConverter converter, IAuthRepository auth, string asUser = null, bool? suppressNotifications = null)
+            : base(config, service, converter, auth, asUser, suppressNotifications) { }
+
+        /// <summary>
+        /// Creates a metadata cascade policy on a folder and the sub folder items. 
+        /// </summary>
+        /// <param name="folderId">The id of the folder to assign the cascade policy to.</param>
+        /// <param name="scope">The scope of the metadata cascade policy.</param>
+        /// <param name="templateKey">The template key of the metadata cascade policy.</param>
+        /// <returns>The metadata cascade policy if successfully created.</returns>
+        public async Task<BoxMetadataCascadePolicy> CreateCascadePolicyAsync(string folderId, string scope, string templateKey)
+        {
+            folderId.ThrowIfNullOrWhiteSpace("folderId");
+            scope.ThrowIfNullOrWhiteSpace("scope");
+            templateKey.ThrowIfNullOrWhiteSpace("templateKey");
+
+            dynamic jsonObject = new JObject();
+            jsonObject.folder_id = folderId;
+            jsonObject.scope = scope;
+            jsonObject.template_key = templateKey;
```

Comment:
This is incorrect — the API endpoint uses the non-standard casing `templateKey` for this parameter